### PR TITLE
Reuse rationalized sig meta

### LINKED ISSUE #1501: Avoid repeating work in handle sig expansion/verification
**Summary**
There are two simple ways to reduce the cost of signature verification in `handleTransaction`; both involve re-using information computed in `Rationalization.execute()`.
- [x] `SwirldTransaction.getSignatures()` acquires a lock and makes a defensive copy; the finalized `List<TransactionSignature>` can be set in a field on the scoped `TxnAccessor`.
- [x] The `AwareProcessLogic` and `InHandleActivationHelper` will _also_ need the payer and other-party required signing keys; these may also be recorded in a `TxnAccessor` field.

### PR DESCRIPTION
**Pull request by tinker-michaelj**
Sonar passed with no non-license code smells [here](https://sonarcloud.io/dashboard?branch=01501-M0150-ReuseRationalizedSigMeta&id=com.hedera.hashgraph%3Ahedera-services).

**Related issue(s)**
- Closes #1501

**Summary of the change**
Some minor code cleanup:
- Remove the `ScopedSigBytesProvider` type, which only existed to hide whether sig bytes were coming from a `SignatureMap` or the now-unsupported `SignatureList`.
- Remove the `SigFactoryCreator` type, which only existed to hide whether an outer txn or a scheduled txn was being signed (and the latter case was eliminated in the scheduled txn refactor).
- Eliminate the unnecessary `ByteString` wrapper from `TxnScopedPlatformSigFactory.create()`.

And a big performance win that avoids two kinds of repeated work being done:
1. The list of required other-party keys was being expanded twice, once in `rationalizeWithPreConsensusSigs()` during `AwareProcessLogic.doProcess()`, and again in `InHandleActivationHelper.areOtherPartiesActive()`.
2. The `SwirldTransaction` list of `TransactionSignatures` was being accessed at least three times: Once to compare the expanded required signing keys during rationalization, once to construct the `Function<byte[], TransactionSignature>` pub-key-to-sig function for testing payer key activation, and yet again to construct the same pub-key-to-sig function for testing other-party key activation.
    - This access is extremely expensive, as it is not only synchronized, but also must return a defensive copy!
    - When the rationalized signatures didn't match the expanded signatures, a fourth access occurred.

The solution in this PR is to create a `RationalizedSigMeta` process object that captures the reusable by-products of `Rationalization.execute()`; that is,
1. The required payer key.
2. The required other-party keys.
3. The canonical list of verified `TransactionSignature`s---no matter whether verified synchronously or asynchronously.
4. The pub-key-to-sig function for testing key activation (backed by the above list).

So in `Rationalization.execute()` we [set a `RationalizedSigMeta` instance](https://github.com/hashgraph/hedera-services/blob/01501-M0150-ReuseRationalizedSigMeta/hedera-node/src/main/java/com/hedera/services/sigs/Rationalization.java#L120) as a field of the `TxnAccessor` in `Rationalization`. This makes reuse trivial, since every downstream component that needs any of the above can just get them from the accessor.

### CODE REVIEW

File: hedera-node/src/main/java/com/hedera/services/sigs/HederaToPlatformSigOps.java [hunk 1]
```diff
@@ -119,24 +116,24 @@ public static SignatureStatus expandIn(
...
 	 * </ul>
 	 *
-	 * @param txnAccessor the accessor for the platform txn.
-	 * @param syncVerifier facility for synchronously verifying a cryptographic signature.
-	 * @param keyOrderer facility for listing Hedera keys required to sign the gRPC txn.
-	 * @param sigsProvider source of crypto sigs for the simple keys in the Hedera key leaves.
+	 * @param txnAccessor the accessor for the platform txn
+	 * @param syncVerifier facility for synchronously verifying a cryptographic signature
+	 * @param keyOrderer facility for listing Hedera keys required to sign the gRPC txn
+	 * @param pkToSigFnProvider source of crypto sigs for the simple keys in the Hedera key leaves
```
Comment (ljianghedera) [inline hedera-node/src/main/java/com/hedera/services/sigs/HederaToPlatformSigOps.java:122]: nit: `@param...`  for `sigFactoryCreator` is missing

Comment (tinker-michaelj) [reply hedera-node/src/main/java/com/hedera/services/sigs/HederaToPlatformSigOps.java:122]: Added!

File: hedera-node/src/main/java/com/hedera/services/utils/RationalizedSigMeta.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
+package com.hedera.services.utils;
+
```
Comment (Neeharika-Sompalli) [inline hedera-node/src/main/java/com/hedera/services/utils/RationalizedSigMeta.java:2]: copyright missing

File: hedera-node/src/main/java/com/hedera/services/utils/RationalizedSigMeta.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import com.hedera.services.legacy.core.jproto.JKey;
+import com.swirlds.common.crypto.TransactionSignature;
+
+import java.util.List;
+import java.util.function.Function;
+
+import static com.hedera.services.keys.HederaKeyActivation.pkToSigMapFrom;
+
```
Comment (Neeharika-Sompalli) [inline hedera-node/src/main/java/com/hedera/services/utils/RationalizedSigMeta.java:10]: javadoc for the class will be helpful

File: hedera-node/src/test/java/com/hedera/services/sigs/RationalizationTest.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
+package com.hedera.services.sigs;
+
```
Comment (Neeharika-Sompalli) [inline hedera-node/src/test/java/com/hedera/services/sigs/RationalizationTest.java:2]: copyright missing

File: hedera-node/src/test/java/com/hedera/services/utils/RationalizedSigMetaTest.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
+package com.hedera.services.utils;
+
```
Comment (Neeharika-Sompalli) [inline hedera-node/src/test/java/com/hedera/services/utils/RationalizedSigMetaTest.java:2]: same here

File: hedera-node/src/main/java/com/hedera/services/utils/RationalizedSigMeta.java (hunk 3 above)
Comment (tinker-michaelj) [reply hedera-node/src/main/java/com/hedera/services/utils/RationalizedSigMeta.java:34]: Added!

File: hedera-node/src/test/java/com/hedera/services/utils/RationalizedSigMetaTest.java (hunk 5 above)
Comment (tinker-michaelj) [reply hedera-node/src/test/java/com/hedera/services/utils/RationalizedSigMetaTest.java:2]: Added :/

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !!

**Pull request merged**